File: frontends/python/ariadne/actions/forge.py
"""Forge (code-hosting platform) semantic actions. The inventory selects the
implementation (e.g. gh, the GitHub CLI)."""

from __future__ import annotations

from typing import Any

from .._impl import semantic, SemanticImpl


def github(
    tag: str | None = None,
    files: list[str] | None = None,
    notes: str | None = None,
) -> SemanticImpl:
    """Create a GitHub release."""
    return semantic("forge.github", tag=tag, files=files, notes=notes)

File: src/lowering/forge.rs
use super::{arg_list, arg_str, def, local, Registry};

/// Forge (code-hosting platform) lowerings.
pub fn register(r: &mut Registry) {
    r.register(def("forge.github.gh", "forge.github", "gh", |a| {
        let tag = arg_str(a, "tag").unwrap_or_default();
        let mut p = vec!["gh".into(), "release".into(), "create".into(), tag];
        p.extend(arg_list(a, "files"));
        local(p)
    }));
}
